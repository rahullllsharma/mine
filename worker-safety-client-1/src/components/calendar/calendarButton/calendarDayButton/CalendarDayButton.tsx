import type { CalendarButtonProps } from "../CalendarButton";
import cx from "classnames";
import { Icon } from "@urbint/silica";
import { getDayAndWeekdayFromDate, checkIsToday } from "@/utils/date/helper";
import { useTenantStore } from "@/store/tenant/useTenantStore.store";
import { RiskLevel } from "../../../riskBadge/RiskLevel";
import CalendarButton from "../CalendarButton";
import Tooltip from "../../../shared/tooltip/Tooltip";
import { getTextColorByRiskLevel } from "../../../../utils/risk";

export type CalendarDayButtonProps = CalendarButtonProps & {
  date: string;
  isCritical?: boolean;
  riskLevel?: RiskLevel;
};

export default function CalendarDayButton({
  date,
  isCritical = false,
  riskLevel = RiskLevel.UNKNOWN,
  ...tailProps
}: CalendarDayButtonProps): JSX.Element {
  const [day, weekday] = getDayAndWeekdayFromDate(date);
  const { activity: activityEntity } = useTenantStore(state =>
    state.getAllEntities()
  );

  const isToday = checkIsToday(date);
  const dayStyles = isToday ? "text-brand-urbint-40" : "text-brand-urbint-60";

  return (
    <CalendarButton {...tailProps} isToday={isToday}>
      <span className={cx("text-base font-semibold", dayStyles)}>{day}</span>
      <div className="flex">
        <span className="mt-1 text-tiny font-medium text-neutral-shade-100">
          {weekday}
        </span>
        {isCritical && (
          <Tooltip
            title={`${activityEntity.attributes.criticalActivity.label} scheduled`}
            position="bottom"
            className="max-w-xm"
          >
            <Icon
              name={"warning"}
              className={cx(
                "pointer-events-none text-lg bg-transparent ml-1 mt-[0.1rem] leading-none",
                getTextColorByRiskLevel(riskLevel)
              )}
            />
          </Tooltip>
        )}
      </div>
    </CalendarButton>
  );
}
