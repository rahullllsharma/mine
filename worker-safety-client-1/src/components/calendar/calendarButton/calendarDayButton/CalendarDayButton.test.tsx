import { render, screen } from "@testing-library/react";
import CalendarDayButton from "./CalendarDayButton";

const mockOnClick = jest.fn();

describe(CalendarDayButton.name, () => {
  it("should render a calendar day button", () => {
    const { asFragment } = render(
      <CalendarDayButton date="2020-01-19" onClick={mockOnClick} />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render with the text matching the day and weekday of the "date" prop', () => {
    render(<CalendarDayButton date="2022-01-20" onClick={mockOnClick} />);
    screen.getByText("20");
    screen.getByText("Thu");
  });

  it("should render a day with a lighter blue color if the date is today", () => {
    const today = new Date().toString();
    const day = new Date(today).toLocaleDateString("en-US", { day: "numeric" });
    render(<CalendarDayButton date={today} onClick={mockOnClick} />);
    const dayElement = screen.getByText(day);
    expect(dayElement).toHaveClass("text-brand-urbint-40");
  });

  it("should render without text content if an invalid date is passed", () => {
    const { asFragment } = render(
      <CalendarDayButton date="invalid date" onClick={mockOnClick} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
