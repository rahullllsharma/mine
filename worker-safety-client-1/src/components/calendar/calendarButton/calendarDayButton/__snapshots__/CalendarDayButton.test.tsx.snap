// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalendarDayButton should render a calendar day button 1`] = `
<DocumentFragment>
  <button
    class="h-20 w-20 flex flex-col items-center justify-center disabled:opacity-30 disabled:cursor-not-allowed border rounded-xl border rounded-xl border-brand-gray-40"
  >
    <span
      class="text-base font-semibold text-brand-urbint-60"
    >
      19
    </span>
    <div
      class="flex"
    >
      <span
        class="mt-1 text-tiny font-medium text-neutral-shade-100"
      >
        Sun
      </span>
    </div>
  </button>
</DocumentFragment>
`;

exports[`CalendarDayButton should render without text content if an invalid date is passed 1`] = `
<DocumentFragment>
  <button
    class="h-20 w-20 flex flex-col items-center justify-center disabled:opacity-30 disabled:cursor-not-allowed border rounded-xl border rounded-xl border-brand-gray-40"
  >
    <span
      class="text-base font-semibold text-brand-urbint-60"
    />
    <div
      class="flex"
    >
      <span
        class="mt-1 text-tiny font-medium text-neutral-shade-100"
      />
    </div>
  </button>
</DocumentFragment>
`;
